pkg/storage/formattedstore/batchwrite.go

```diff
@@ -35,6 +35,7 @@ type BatchService struct {
 	formatter        Formatter
 	provider         batchProvider // TODO remove batch provider after refactoring edv batch
 	current          *pendingBatch
+	batchSizeLimit   int
```

Comment:
We could also additionally consider an optional max batch time in a followup.

Reply:
yes